extensions/resource-deployment/src/services/tools/SemVerProxy.ts

```diff
@@ -0,0 +1,55 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { SemVer } from 'semver';
+
+
+
+function fourPart2SemVer(version: string): string {
+	if (version.includes('-')) {
+		//return unchanged if version contains a pre-release suffix
```

Comment:
According to the spec a valid SemVer can contain a build part in the pre-release suffix. Are you sure that the version string we get isn't going to possible look something like this?

`15.0.2070-prerelease.1`

and thus be invalid syntax (since it expects a + instead of a . there)

Reply:
This is unknown, the current 4 part version is motivated by SQL versions which do not have those parts. So instead of design for what might be in the future, we will do a fix/change when we get there. For now, the idea was to allow the '4th' build part separated by '.' only in the primary version string.

Reply:
not currently as the 4 part version string is motivated by the SQL version string numbering scheme. So if something like this comes about in the future we will tackle it then.